include/mem.h

```diff
@@ -159,11 +109,104 @@ constexpr static INLINE uint32_t host_to_le(uint32_t val) {
 	return val;
 }
 
+constexpr static INLINE uint64_t host_to_le(uint64_t val)
+{
+	return val;
+}
+
 #endif
 
-static INLINE void var_write(Bit8u * var, Bit8u val) {
-	host_writeb(var, val);
+constexpr static INLINE uint8_t le_to_host(uint8_t val)
+{
+	return host_to_le(val);
+}
+
+constexpr static INLINE int16_t le_to_host(int16_t val)
+{
+	return host_to_le(val);
+}
+
+constexpr static INLINE uint16_t le_to_host(uint16_t val)
+{
+	return host_to_le(val);
+}
+
+constexpr static INLINE uint32_t le_to_host(uint32_t val)
+{
+	return host_to_le(val);
+}
+
+constexpr static INLINE uint64_t le_to_host(uint64_t val)
+{
+	return host_to_le(val);
+}
+
+// Read, write, and add using 16-bit words
+static INLINE uint16_t host_readw(const uint8_t *arr)
+{
+	uint16_t val;
+	memcpy(reinterpret_cast<void *>(&val),
+	       reinterpret_cast<const void *>(arr), sizeof(val));
```

Comment:
Actually, I'm not sure if we should use `reinterpret_cast` here? Maybe we're making it harder for the compiler to optimize the code if we remove type information? (considering compilers notice it's a memcpy call and try to optimize it)

`void*` type is "special" in C and C++, I think we can write this part:

`memcpy(&val, arr, sizeof(val));`

(constness of our pointers matches constness of memcpy params, so we should be fine)

That being said, I know MSVC has problems with properly casting void pointers in certain contexts (when casting arrays or pointers to pointers) - but in this case I think it should work fine.

(same comment for other places where use reinterpret_cast for memcpy call)

Reply:
Was worried about warnings due to not matching it's desired void* arg types, but this doesn't seem to be an issue; still building clean.